signac/common/config.py

```diff
@@ -12,18 +12,19 @@
 
 logger = logging.getLogger(__name__)
 
-DEFAULT_FILENAME = ".signacrc"
-CONFIG_FILENAMES = [DEFAULT_FILENAME, "signac.rc"]
-HOME = os.path.expanduser("~")
-CONFIG_PATH = [HOME]
-FN_CONFIG = os.path.expanduser("~/.signacrc")
+# TODO: Should these variables be internal or do we want them exposed?
+PROJECT_CONFIG_FN = os.path.join(".signac", "config")
+USER_CONFIG_FN = os.path.expanduser("~/.signacrc")
```

Comment:
Pendantic:
```suggestion
USER_CONFIG_FN = os.path.expanduser(os.path.join("~", ".signacrc"))
```

Reply:
I can confirm this is needed on Windows:
```python
>>> os.path.expanduser("~/.signacrc")
'C:\\Users\\bdice/.signacrc'
>>> os.path.expanduser(os.path.join("~", ".signacrc"))
'C:\\Users\\bdice\\.signacrc'
```

As a note, these lines will execute any time the package is imported. We want to minimize the syscall requirements to keep fast import times (all the `os` functions here must perform syscalls, to my knowledge). It may be more appropriate to define anything calling `os.path.join` or `os.path.expanduser` in a function that is called when needed, rather than as a module constant evaluated at import time. It looks like these module constants are used O(1) times and not O(N) times, so it's not paying off the cost incurred at import time, only shifting a cost from usage-time to import-time.

Reply:
> I can confirm this is needed on Windows:

I had a hunch that this might be an issue on Windows, but figured that it wasn't an actual bug, because it had not been an issue until now. 😕